src/web/src/components/Posts/PostAvatar.tsx

```diff
@@ -0,0 +1,50 @@
+import { createStyles, Theme, Avatar } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+
+type AvatarProps = {
+  name: String;
+  img?: String;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    avatar: {
+      marginLeft: '0.5rem',
+      color: '#e5e5e5',
+      backgroundColor: '#121D59',
+      fontSize: '2.5rem',
+      width: '3em',
+      height: '3em',
+      [theme.breakpoints.down(1200)]: {
+        fontSize: '2em',
+        width: '2.5em',
+        height: '2.5em',
+      },
+    },
+  })
+);
+
+const PostAvatar = ({ name, img }: AvatarProps) => {
+  const classes = useStyles();
+
+  if (img) {
+    return <Avatar className={classes.avatar} src="img" />;
+  }
+
+  if (name) {
+    const nameArr = name.split(' ');
+    let initials;
```

Comment:
I wonder if we could simplify this a bit.  Something like:

```js
const initials = name.split(' ').map(word => word[0].toUpperCase()).join('').slice(0,2);
```

This should work for any name.  Question: do we want to split on space and/or dash?

Reply:
Currently there is no space between the initials, below are initials without anything, with dot, with dash, with space.

![Screenshot_2021-03-12 Telescope(4)](https://user-images.githubusercontent.com/50813726/110972169-b1834500-8329-11eb-8c72-39ac5d7c0374.png)
![Screenshot_2021-03-12 Telescope(3)](https://user-images.githubusercontent.com/50813726/110972171-b1834500-8329-11eb-999e-1654e0078403.png)
![Screenshot_2021-03-12 Telescope(2)](https://user-images.githubusercontent.com/50813726/110972172-b21bdb80-8329-11eb-9236-589f4de10c71.png)
![Screenshot_2021-03-12 Telescope(1)](https://user-images.githubusercontent.com/50813726/110972270-cd86e680-8329-11eb-99f2-b29df2afc5c2.png)

Reply:
I don't mean we should include them in the final version, I more mean should we split on `/[ -]/`.  For example, what would you expect your name to be? YH or YL?  I agree that the final form should be 2 characters right beside each other (no space or dashes).

Reply:
Ah I see, sorry for the misunderstanding.

I prefer split names by space, to my understanding, dash is more like linking? In my case, Yuan-Hsi is my first name, and everyone from Taiwan is using dash to link our 2-syllable first name. So YL makes much sense.

But I'm not sure which one is better.

Reply:
I think you've answered it: we should just split on spaces, and assume a dash is a linked name.  Let's do that.

Reply:
> I wonder if we could simplify this a bit. Something like:
> 
> ```js
> const initials = name.split(' ').map(word => word[0].toUpperCase()).join('').slice(0,2);
> ```
> 
> This should work for any name. Question: do we want to split on space and/or dash?

Also, another question, should the initials take the first 2 words' first letters, or the first and the last words' letters?

For example, name "Yuri Joe Chapman" should return YC or YJ? I believe it's YC, but the suggested code from your comment will return YJ instead. I'm not sure which one is preferred?

Reply:
Ah, good point.  Yeah, modify my code to do first and last.

Reply:
@humphd just pushed new commit, thank you!